### PR TITLE
Add file size task

### PR DESCRIPTION
**Pull request by Landerstraeten**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | #374 

# New Task Checklist:

- [x] Is the README.md file updated?
- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpspec tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

File: spec/Task/FileSizeSpec.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        $context->getFiles()->willReturn(new FilesCollection([
+            new SplFileInfo('src/Collection/TaskResultCollection.php', 'src/Collection', 'TaskResultCollection.php'),
+        ]));
+
+        $result = $this->run($context);
+        $result->shouldBeAnInstanceOf(TaskResultInterface::class);
+        $result->isPassed()->shouldBe(true);
+    }
+}
```
Comment (veewee) [inline spec/Task/FileSizeSpec.php]: Can you add a tests for the failing task?

File: src/Task/FileSize.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'max_size' => '10M',
+        ]);
+
+        $resolver->addAllowedTypes('max_size', ['string']);
```
Comment (veewee) [inline src/Task/FileSize.php]: It is also possible to just add a number in bytes without a suffix. (see \Symfony\Component\Finder\Comparator\NumberComparator)
So maybe it should also be possible to allow ints?

File: src/Task/FileSize.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'max_size' => '10M',
```
Comment (veewee) [inline src/Task/FileSize.php]: Is there a specific reason you've choosen 10M here?

**Comment by veewee** [comment]
@Landerstraeten,

Thanks for the PR! Looks good but I've added some small comments / questions.

File: src/Task/FileSize.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        return $this->getConfigurableOptions()->resolve($configured);
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
```
Comment (veewee) [inline src/Task/FileSize.php]: Maybe it's a good idea to add a whitelist or blacklist  like e.g. `triggered_by` which whitelists extensions. This way you can still allow for example PDFs, ... to be committed.

File: src/Task/FileSize.php (hunk 3 above)
Comment (Landerstraeten) [inline src/Task/FileSize.php]: I have based this on SourceTree which gives a warning for files over 10M.
Github gives a warning for files over 50M and restricts files over 100M.
What do you suggest?

File: spec/Task/FileSizeSpec.php (hunk 1 above)
Comment (Landerstraeten) [inline spec/Task/FileSizeSpec.php]: What do you mean with the failing task? What test do I need to write exactly?

Comment (veewee) [inline spec/Task/FileSizeSpec.php]: A test that checks the failed return when a file is bigger than the configured max_size.

File: src/Task/FileSize.php (hunk 3 above)
Comment (veewee) [inline src/Task/FileSize.php]: Let's keep it at 10M for now.

File: src/Task/FileSize.php (hunk 4 above)
Comment (lykciv) [inline src/Task/FileSize.php]: I'm not sure a whitelist has an added value in this case. Big files will cause problems in git regardless of file type. If you would whitelist PDF files, someone could still commit a 500MB PDF. 

If you have multiple bigger files in your repository, you can always set the max_size higher. And if you find that you have to set the max_size too high, that's probably a sign that you should review whether git is the right solution for storing your bigger files.

**Comment by veewee** [comment]
Thanks for the PR @Landerstraeten !

**Pull request merged**